apps/dao/src/store/createProposalsSlice.ts

```diff
@@ -477,19 +477,50 @@ const createProposalsSlice = (set: SetState<State>, get: GetState<State>): Propo
 
           await helpers.waitForTransaction(voteResponseHash, provider)
 
-          get()[sliceKey].setStateByKey('executeTxMapper', {
-            ...get()[sliceKey].executeTxMapper,
-            [voteIdKey]: {
-              status: 'SUCCESS',
-              hash: voteResponseHash,
-              txLink: networks[1].scanTxPath(voteResponseHash),
-              error: null,
-            },
-          })
-
           dismissDeploying()
           const successNotificationMessage = t`Proposal executed successfully!`
           notifyNotification(successNotificationMessage, 'success', 15000)
+
+          // update proposal executed status
+          const response = await fetch(
```

Comment:
Can't you just call `getProposal()` again to fetch fresh data? It feels as if this is code duplication at this point. Alternatively, just add an optional `tx_hash` parameter to `getProposal()`.

Reply:
I could, I don't want to initiate a loading state so this runs in the background without triggering any spinners but that can also be optionalised ofc

Reply:
Yea, this is a good suggestion. I wanted to do a silent refetch of data but can do it within the getProposal method.